gap/Forms.gi

```diff
@@ -0,0 +1,157 @@
+# If <type> = "S" or type = "O" then <group> must have the attribute
+# InvariantBilinearForm.
+InstallGlobalFunction("ChangeFixedSesquilinearForm",
+function(group, type, gramMatrix)
+    local gapForm, newForm, gapToCanonical, canonicalToNew, field, formMatrix;
+    if not type in ["S", "O", "U"] then
+        ErrorNoReturn("<type> must be one of 'S', 'U', 'O', but <type> = ",
+                      type);
+    fi;
+    field := DefaultFieldOfMatrixGroup(group);
+    if type = "S" or type = "O" then
+        gapForm := BilinearFormByMatrix(InvariantBilinearForm(group).matrix, 
+                                        field);
+        newForm := BilinearFormByMatrix(gramMatrix, field);
+    else
+        if HasInvariantSesquilinearForm(group) then
+            gapForm := HermitianFormByMatrix(InvariantSesquilinearForm(group).matrix,
+                                             field);
+        else    
+            formMatrix := UnitaryForm(group);
+            if formMatrix = fail then
+                Error("No preserved unitary form found for <group>");
+            fi;
+            gapForm := HermitianFormByMatrix(formMatrix, field);
+        fi;
+        newForm := HermitianFormByMatrix(gramMatrix, field);
+    fi;
+    # the following if condition can only ever be fulfilled if <group> is an
+    # orthogonal group; there the case of even dimension is problematic since,
+    # in that case, there are two similarity classes of bilinear forms
+    if not WittIndex(gapForm) = WittIndex(newForm) then
+       ErrorNoReturn("The form preserved by <group> must be similar to the form ", 
+                     "described by the Gram matrix <gramMatrix>.");
+    fi;
+    gapToCanonical := BaseChangeHomomorphism(BaseChangeToCanonical(gapForm), 
+                                             field);
+    canonicalToNew := BaseChangeHomomorphism(BaseChangeToCanonical(newForm) ^ (-1), 
+                                             field);
+    return Group(canonicalToNew(gapToCanonical(GeneratorsOfGroup(group))));
+end);
+
+# Can only deal with sesquilinear forms, not with quadratic forms as of yet.
+# If <type> is one of "S", "O+", "O-" or "O" then <group> must have the
+# attribute InvariantBilinearForm.
+InstallGlobalFunction("ConjugateToGAPForm",
```

Comment:
I suggest `ConjugateToStandardForm`. I think it makes sense to document this. Maybe this should go into the gap library? :thinking:

Reply:
Can't, is requires the forms package. But perhaps could be added there.

Reply:
See #40.